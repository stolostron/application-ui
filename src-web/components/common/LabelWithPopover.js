// Copyright (c) 2020 Red Hat, Inc.
// Copyright Contributors to the Open Cluster Management project
'use strict'

import React from 'react'
import PropTypes from 'prop-types'
import {
  Divider,
  Label,
  Popover,
  Stack,
  StackItem
} from '@patternfly/react-core'
import resources from '../../../lib/shared/resources'

resources(() => {
  require('../../../scss/label-with-popover.scss')
})

const LabelWithPopover = ({
  children,
  labelContent,
  labelIcon,
  labelColor
}) => (
  <div className="label-with-popover">
    <Popover
      bodyContent={
        <Stack>
          <StackItem className="close-button-spacer" />
          <StackItem>
            <Divider />
          </StackItem>
          <StackItem>{children}</StackItem>
        </Stack>
      }
      className="label-with-popover"
      enableFlip
      position="bottom"
      flipBehavior={['bottom', 'top', 'right', 'left']}
      distance={10}
      zIndex={999}
    >
      <Label
        onClick={event => {
          event.preventDefault()
          event.nativeEvent.preventDefault()
          event.stopPropagation()
        }}
        color={labelColor ? labelColor : 'grey'}
        href="#"
        icon={labelIcon}
      >
        {labelContent}
      </Label>
    </Popover>
  </div>
)

LabelWithPopover.propTypes = {
  children: PropTypes.object,
  labelColor: PropTypes.string,
  labelContent: PropTypes.object,
  labelIcon: PropTypes.object
}

export default LabelWithPopover
