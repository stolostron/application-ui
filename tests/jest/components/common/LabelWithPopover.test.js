/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2019. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
"use strict";

import React from "react";
import { Icon } from "carbon-components-react";
import LabelWithPopover from "../../../../src-web/components/common/LabelWithPopover";
import renderer from "react-test-renderer";

describe("LabelWithPopover", () => {
  it("renders as expected", () => {
    const component = renderer.create(
      <LabelWithPopover
        labelIcon={<Icon name="icon--launch" />}
        labelContent="TheLabel"
      >
        ThePopoverContent
      </LabelWithPopover>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
