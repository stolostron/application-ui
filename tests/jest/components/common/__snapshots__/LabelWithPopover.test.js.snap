// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabelWithPopover renders as expected 1`] = `
<div
  className="label-with-popover"
>
  <span
    className="pf-c-label"
    onClick={[Function]}
    onPointerEnter={[Function]}
    onPointerLeave={[Function]}
  >
    <a
      className="pf-c-label__content"
      href="#"
    >
      <span
        className="pf-c-label__icon"
      >
        <svg
          alt="Provide a description that will be used as the title"
          aria-label="Provide a description that will be used as the title"
          fillRule="evenodd"
          height="16"
          name="icon--launch"
          role="img"
          viewBox="0 0 16 16"
          width="16"
        >
          <title>
            Provide a description that will be used as the title
          </title>
          <path
            d="M14.3 1h-3.8V0H16v5.5h-1V1.7L9.7 7 9 6.3 14.3 1z"
          />
          <path
            d="M14.3 1h-3.8V0H16v5.5h-1V1.7L9.7 7 9 6.3 14.3 1z"
          />
          <path
            d="M13 9h1v6c0 .6-.4 1-1 1H1c-.6 0-1-.4-1-1V3c0-.6.4-1 1-1h7v1H1v12h12V9z"
          />
        </svg>
      </span>
      TheLabel
    </a>
  </span>
</div>
`;
